### PR TITLE
add prisma support

### PR DESCRIPTION
**Pull request by tingfeng-key**
add prisma support

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 
 ### Features/Changes
+- [#1700](https://github.com/lapce/lapce/pull/1700):add prisma support
```
Comment (panekj) [inline CHANGELOG.md:6]: missing space after `:`

File: lapce-core/queries/prisma/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,56 @@
+(string) @string
+
+(enumeral) @constant
+(number) @constant.numeric
+
+(variable) @variable
```
Comment (panekj) [inline lapce-core/queries/prisma/highlights.scm:6]: Where is the file from?

Comment (tingfeng-key) [reply lapce-core/queries/prisma/highlights.scm:6]: here: [https://github.com/victorhqc/tree-sitter-prisma/blob/master/queries/highlights.scm](https://github.com/victorhqc/tree-sitter-prisma/blob/master/queries/highlights.scm)

Comment (panekj) [reply lapce-core/queries/prisma/highlights.scm:6]: Please include that information like other queries do

Comment (tingfeng-key) [reply lapce-core/queries/prisma/highlights.scm:6]: Oh, it should be written in the head of the file, right?

Comment (tingfeng-key) [reply lapce-core/queries/prisma/highlights.scm:10]: OK, it has been added. I misunderstood

File: lapce-core/src/language.rs [hunk 3]
```diff
@@ -210,6 +210,8 @@ pub enum LapceLanguage {
     Yaml,
     #[cfg(feature = "lang-zig")]
     Zig,
+    #[cfg(feature = "lang-prisma")]
+    Prisma,
```
Comment (panekj) [inline lapce-core/src/language.rs:214]: just noticed that it isn't sorted alphabetically

**Review by panekj — CHANGES_REQUESTED** [review]

File: lapce-core/Cargo.toml [hunk 4]
```diff
@@ -125,3 +126,4 @@ lang-yaml = ["dep:tree-sitter-yaml"]
 lang-xml = ["dep:tree-sitter-xml"]
 lang-zig = ["dep:tree-sitter-zig"]
 lang-sql = ["dep:tree-sitter-sql"]
+lang-prisma = ["dep:tree-sitter-prisma-io"]
```
Comment (panekj) [inline lapce-core/Cargo.toml:129]: this should also be sorted

File: lapce-core/Cargo.toml [hunk 5]
```diff
@@ -73,6 +73,7 @@ tree-sitter-yaml = { git = "https://github.com/panekj/tree-sitter-yaml", branch
 tree-sitter-zig = { git = "https://github.com/maxxnino/tree-sitter-zig", branch = "main", optional = true }
 # switch to upstream version after this is merged  https://github.com/m-novikov/tree-sitter-sql/pull/68
 tree-sitter-sql = { git= "https://github.com/oknozor/tree-sitter-sql", branch ="main", optional = true }
+tree-sitter-prisma-io = {version = "1.3.0", optional = true}
```
Comment (panekj) [inline lapce-core/Cargo.toml:76]: this should also be sorted

File: lapce-core/src/language.rs [hunk 6]
```diff
@@ -849,6 +851,18 @@ const LANGUAGES: &[SyntaxProperties] = &[
...
+        language: tree_sitter_prisma_io::language,
+        highlight: include_str!("../queries/prisma/highlights.scm"),
+        injection: None,
+        comment: "//",
+        indent: "    ",
+        code_lens: (DEFAULT_CODE_LENS_LIST, DEFAULT_CODE_LENS_IGNORE_LIST),
+        sticky_headers: &[],
+        extensions: &["prisma"],
+    },
 ];
```
Comment (panekj) [inline lapce-core/src/language.rs:866]: this should also be sorted

File: lapce-core/src/syntax/highlight.rs [hunk 7]
```diff
@@ -102,6 +102,7 @@ declare_language_highlights!(
     Xml: "lang-xml",
     Yaml: "lang-yaml",
     Zig: "lang-zig",
+    Prisma: "lang-prisma",
```
Comment (panekj) [inline lapce-core/src/syntax/highlight.rs:105]: this should also be sorted

File: lapce-ui/Cargo.toml [hunk 8]
```diff
@@ -95,6 +95,7 @@ all-languages = [
     "lang-yaml",
     "lang-zig",
     "lang-sql",
+    "lang-prisma",
```
Comment (panekj) [inline lapce-ui/Cargo.toml:98]: this should also be sorted

File: lapce-ui/Cargo.toml [hunk 9]
```diff
@@ -143,3 +144,4 @@ lang-wgsl = ["lapce-core/lang-wgsl"]
 lang-xml = ["lapce-core/lang-xml"]
 lang-yaml = ["lapce-core/lang-yaml"]
 lang-zig = ["lapce-core/lang-zig"]
+lang-prisma = ["lapce-core/lang-prisma"]
```
Comment (panekj) [inline lapce-ui/Cargo.toml:147]: this should also be sorted

**Review by panekj — CHANGES_REQUESTED** [review]

**Comment by tingfeng-key** [comment]
When I make local modifications, cargo.toml changes after running fmt. I don't know how those are merged. I think this is somewhat targeted

**Comment by tingfeng-key** [comment]
This makes me take it for granted that the latest should be placed at the bottom. I apologize

**Pull request merged**